### DISCUSSION TITLE
Language.from_bytes creating a blank model

### DISCUSSION THREAD

**Discussion by DarrelYee-SiA** [explosion/spaCy#9153] [Help: Coding & Implementations] [answered]
This issue was raised a while back in #1942  but was closed without resolution.

Serializing a model through Language.to_bytes() then de-serializing it through Language.from_bytes() creates a blank model instead of the original model. 

## How to reproduce the behaviour
The following code replicates the issue:

```
# Load en_core_web_sm as an example.
nlp1 = spacy.load('en_core_web_sm')
ser = nlp1.to_bytes()

# Start with blank model, then de-serialize.
nlp2 = spacy.blank('en')
nlp2=nlp2.from_bytes(ser)
nlp2.to_disk('test_mod')
```

This code creates a blank model with no components in the local directory.

## Your Environment
**spaCy version:** 3.1.2
**Platform:** Windows-10-10.0.19043-SP0
**Python version:** 3.9.6

**Comment by polm** [comment]
Thanks for the report. I was able to confirm this behavior.

It looks like the issue is that instead of reading the list of components from the serialized bytes, the list of components in the current pipeline is used as-is. 

https://github.com/explosion/spaCy/blob/ba6a37d3588f3d532733e5f2df449aeecd3f47fa/spacy/language.py#L2020

It looks like in order for this to work as expected we would have to do something closer to what happens in `spacy.load`, which involves `load_model_from_config` to set up the pipeline correctly.

Also note that while this looks like a bug to me, it's also not something we typically imagine you having to do - in general, the right way to serialize a model is to use `nlp.to_disk` and load it with `spacy.load`.

**Comment by DarrelYee-SiA** [comment]
Hi, thanks for looking into this.

Our use case requires the spacy model to be hosted as a single byte array to our postgres server; having multiple components from `Language.to_disk()` is inconvenient, which is why I was looking into `Language.to_bytes()` as a solution.

**Comment by adrianeboyd** [comment, marked as answer]
Sorry, let me jump in here:

While I would agree that this is a bit confusing/unintuitive given the parallel-looking method names `to_bytes` / `from_bytes`, this is the intended behavior for `from_bytes` for a pipeline and is not a bug.

Here are the steps required along with `nlp.to_bytes` / `nlp.from_bytes` to save and restore the full pipeline: https://spacy.io/usage/saving-loading/#pipeline
